### ISSUE TITLE
Option for single pull request per update

### ISSUE THREAD

**Issue by greysteil** [dependabot/dependabot-core#369]
_From @zach-taylor on June 20, 2017 18:37_

We use deppbot (https://www.deppbot.com) at the moment and we like the idea of a single pull request for each update rather than a pull request for each gem update. The main reason we like this is we have a lot of dependencies, and it can get noisy and slow to perform a build for each one and the way our branch protections are set up.

Thanks for the consideration!

_Copied from original issue: dependabot/feedback#11_

**Comment by greysteil** [comment]
Thanks Zach. We might add a setting for this at some point, but it's not on our roadmap right now - we prefer the multiple PRs approach as it makes them easier to test and merge. That said, we can see a world where we open a single PR for several dependency updates we know are backwards compatible.

In the meantime, [Deppbot](https://www.deppbot.com) are the guys to use if you want a single PR with many Ruby updates in it.

**Comment by greysteil** [comment]
_From @zach-taylor on June 21, 2017 18:55_

Sounds good 👍 I figured I would put it out there and explain our reasoning.

**Comment by greysteil** [comment]
Quick update: we'd still love to do this (group non-breaking updates), but it's going to need us to have lots of data to do it well (we need to know which changes are breaking). One for the future - don't worry, it won't fall off our roadmap.

**Comment by greysteil** [comment]
_From @ad-m on October 1, 2017 12:19_

In my opinion it is not a problem that many changes will be under one pull request. Many CIs test each commit separately, so you can continue to accurately identify the defective dependency.

**Comment by greysteil** [comment]
_From @elamperti on November 5, 2017 2:32_

+1 to this request. Having a weekly/monthly update of all my npm packages would allow me to check quickly if they break something. If that's the case, I can analyze it and fix it manually, but otherwise it's faster than the time and resource consuming alternative of bumping versions for each package. I want to focus on the code, not on updates, and one PR per package creates a lot of noise -at least in my small repos-.

**Comment by greysteil** [comment]
_From @shusson on November 20, 2017 5:59_

@greysteil any news on this? I would like to be able to bump with one PR, if it breaks, bump with multiple PRs. The hope being that the original PR doesn't break often and thus we get the best of both flows.

**Comment by greysteil** [comment]
We're still a way off, but should get closer in the next few weeks. Basically, I need to do a lot of refactoring because everywhere in the update process it expects to be dealing with a single dependency. I should be able to get to that pretty soon, and this isn't the only thing waiting on it. Once that's done, the changes to make this possible are a *lot* more straightforward.

In the meantime, I continue to recommend [Deppbot](https://deppbot.com) if you want a single PR with many Ruby updates in it 🙂 .

**Comment by greysteil** [comment]
_From @styfle on December 9, 2017 17:1_

This was pretty painful for my Electron project to build the PR and then build the build `master` after merging. https://github.com/styfle/magnemite

I think the big problem is that after accepting 1 PR, all the other PRs need to be rebased and rebuilt again. So for 6 deps that need to be updated, you have to wait for 21 builds to complete. At 5 minutes per build, thats almost 2 hours. And you are spending that time clicking waiting and watching CI so you can click Merge on the next PR.

I did [fix a bug](https://github.com/styfle/react-server-example-tsx/pull/32) with travis-ci that was building the PR twice, once for the branch and once for the PR. But this is the default behavior so I'm probably not the only one with that configuration.

**Comment by mikeckennedy** [comment]
+1 for this feature. I would much prefer a single PR per week to group all the updates. It's easy enough to diagnose this manually if something is wrong. Why make the case that works 99% of time harder? I could always be a config option off by default.

**Comment by DavG** [comment]
+1
At least add some rules to "group" updates.
We use symfony framework and (almost) all packages versions are synchronized.
It's useless and weird having updates and seperated PR : 

I just had the case : dependabot updated a symfony package to 4.2.12 and updated another package to 4.3.9. Not logical in that case.

**Comment by flavioribeiro** [comment]
+1

**Comment by samhwang** [comment]
+1

This would be a very nice addition to dependabot's arsenal.

**Comment by mkay581** [comment]
@greysteil my team is looking for grouping of PRs of the _same_ dependency. Often times we may leave a dependabot PR sitting because we need to do some investigation and we'll leave comments on the PR. But a newer version of the dependency comes out which makes dependabot close the original PR losing all of our comments. Is this related to this issue or should a new one be created?

**Comment by greysteil** [comment]
I don't work on Dependabot anymore, but @feelepxyz and the team may be able to help.

**Comment by feelepxyz** [comment]
> my team is looking for grouping of PRs of the _same_ dependency. Often times we may leave a dependabot PR sitting because we need to do some investigation and we'll leave comments on the PR. But a newer version of the dependency comes out which makes dependabot close the original PR losing all of our comments. Is this related to this issue or should a new one be created?

This could probably be achieved if we support configurable grouping. Currently, we open a new PR for every update because we rely on things like the branch name to be unique per version update and this would no longer apply once we start updating an existing PR with newer versions.

Support for grouping updates is high up our list: https://github.com/dependabot/feedback/issues/5

We're currently busy integrating Dependabot natively in GitHub so we've paused on feature work for the next few months while we complete this work.

**Comment by Br4nd0R** [comment]
+1 for this feature

**Comment by drtyh2o** [comment]
Would be super cool dependabot would group updated dependencies from the same scope into a single PR. Often, these scoped packages are published as a set and need to be updated as such.

**Comment by CarlosHAraujo** [comment]
+1

**Comment by bwoodsend** [comment]
Can we at least reopen this issue since it's evidently not completed?

**Comment by eseglem** [comment]
Leaving this here in case anyone else ends up here from Google, like I did, disappointed to see it closed. 

There is a similar open issue which should help address this: https://github.com/dependabot/dependabot-core/issues/1190 and currently looks like it is on the  Q2 2023 roadmap https://github.com/github/roadmap/issues/148

**Comment by jeffwidman** [comment]
Most of the feature requests I see here are covered under the various tickets tracking grouped updates:
https://github.com/dependabot/dependabot-core/labels/F%3A%20grouped-updates%20%F0%9F%A7%91%E2%80%8D%F0%9F%A4%9D%E2%80%8D%F0%9F%A7%91

So I'm going to leave closed as a duplicate of one or more of those linked issues.

**Comment by bwoodsend** [comment]
Looks like #1190 is the one to watch instead.

**Comment by DaleMckeown** [comment]
+1 for this feature request.

**Comment by mnik01** [comment]
+1

**Comment by The-Compiler** [comment]
Please [use +1 reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) if you want to +1 something, that's what they are here for.

I'm assuming dozens of people are subscribed to threads like this, so you're sending everyone a notification containing a "+1" comment, which is... not exactly useful.

**Comment by jeffwidman** [comment]
Thanks @The-Compiler !

We even have it documented as a policy we endorse in this repo: https://github.com/dependabot/dependabot-core#no-1-comments
